firmware/boron-ins-fsm/src/stateMachine.cpp

```diff
@@ -260,26 +266,29 @@ void state3_stillness(){
     Log.warn("door opened, session over, going from state3_stillness to idle");
     publishStateTransition(3, 0, checkDoor.doorStatus, checkINS.iAverage);
     saveStateChange(3, 2);
+    hasDurationAlertBeenSent = 0; // session is over, so reset flag
     stateHandler = state0_idle;
-
   }
-  else if(millis() - state2_duration_timer >= state2_max_duration){
+  else if((millis() - state2_duration_timer >= state2_max_duration)
+          && !hasDurationAlertBeenSent){
 
     Log.warn("See duration alert, going from state3 to idle after alert publish");
     publishStateTransition(3, 0, checkDoor.doorStatus, checkINS.iAverage);
     saveStateChange(3, 4);
     Log.error("Duration Alert!!");
     Particle.publish("Duration Alert", "duration alert", PRIVATE);
-    stateHandler = state0_idle;
+    hasDurationAlertBeenSent = 1; // a duration alert has been sent, so do not send another one
+    stateHandler = state2_duration;
   }
   else if(millis() - state3_stillness_timer >= state3_max_stillness_time){
 
-    Log.warn("stillness alert, going from state3 to idle after publish");
-    publishStateTransition(3, 0, checkDoor.doorStatus, checkINS.iAverage);
+    Log.warn("stillness alert, going from state3 to state2 after publish");
+    publishStateTransition(3, 2, checkDoor.doorStatus, checkINS.iAverage);
```

Comment:
It looks like the way you've implemented this, you actually stay in state 3, so this needs to be updated or removed.

Reply:
Good catch. I updated the log and removed the state transition publish

Reply:
Do you think that we should keep the saveStateChange() here? We publish an alert, but do not change states - please let me know what you think. I personally think we should remove or modify it.

Reply:
I actually think that we should keep the `saveStateChange` call. It's unfortunately named now that we are no longer changing state, but its purpose is to add the state change and its reason to the queue of changes that are sent to the server in the next heartbeat message. It's useful to have all the Stillness and Duration alerts appear as reasons in the heartbeat, so even though the state doesn't change.

Reply:
I'd be open to changing the name of the `saveStateChange` function to something more accurate now. However, the JSON key `state` that is sent as part of the heartbeat will be a pain to change because the server is using it to know what to store in the DB. Up to you.